aiosmtpd/smtp.py

```diff
@@ -535,14 +543,9 @@ def smtp_DATA(self, arg):
         received_data = EMPTYBYTES.join(data)
         if self._decode_data:
             received_data = received_data.decode('utf-8')
-        args = (self.peer, self.mailfrom, self.rcpttos, received_data)
-        kwargs = {}
-        if not self._decode_data:
-            kwargs = {
-                'mail_options': self.mail_options,
-                'rcpt_options': self.rcpt_options,
-                }
-        kwargs.update({'loop': self.loop})
+        self.message.received_data = received_data
+        args = (self.session, self.message)
+        kwargs = {'loop': self.loop}
```

Comment:
I'm thinking that `loop` is the only keyword argument we'll still need.  That's an artifact of the `asyncio` heritage, so it should not go in the session object.